File: problem2.js
function fiboEvenSum(n) {
  
  var fibSeries = [];
  fibSeries[0] = 1;
  fibSeries[1] = 2;
  var ctr = 2
  for (var i=3; i<=n; i++) {
    fibSeries[ctr] = fibSeries[ctr-1]+fibSeries[ctr-2];
    ctr++;
  }
  //console.log("Fibonacci series is "+fibSeries)

  var sum = 0;
  for (var j=0; j< fibSeries.length; j++) {
    if ( fibSeries[j] % 2 == 0) {
      sum = sum + fibSeries[j];
    }
  }
  //console.log("Sum is "+sum)
  return sum;
}

fiboEvenSum(23);

File: go/problem2.go
package main

import (
	"fmt"
)

func main() {

	fmt.Println(SumOfEvenFibonacciNumber(10))
}

func SumOfEvenFibonacciNumber(limit int) int {

	var fibSeries = make([]int, limit)
	fibSeries[0] = 1
	fibSeries[1] = 2

	for counter := 2; counter < limit; counter++ {
		if fibSeries[counter] > limit {
			break
		}
		fibSeries[counter] = fibSeries[counter-1] + fibSeries[counter-2]
	}
	//fmt.Println("Value of Fibonacci Sereis is ", fibSeries)

	sum := 0
	for j := 0; j < len(fibSeries); j++ {
		if fibSeries[j] > limit {
			break
		}
		if fibSeries[j]%2 == 0 {
			sum = sum + fibSeries[j]
		}
	}
	//fmt.Println("Sum of Even numbers is ", sum)
	return sum
}
